src/iframe-helper.js

```diff
@@ -380,3 +380,61 @@ function parseIfNeeded(data) {
   }
   return data;
 }
+
+
+
+/**
+ * Manages a postMessage API for an iframe with a subscription message and
+ * a way to broadcast messages to all subscribed windows, which
+ * in turn must all be descendants of the contentWindow of the iframe.
+ */
+export class SubscriptionApi {
```

Comment:
@nekodo apologize. I obviously had a misunderstanding of how the `requestCallback` was used. I thought the `send` function is passed back as the `requestCallback`, that's why i think they should be merged into one function,  and the `clientWindows_` can be a local variable.

Yes, the implementation with class is much more readable than the one with function.
Thanks for clarifying.